src/lambda-handler.js

```
@@ -0,0 +1,68 @@
+import env from './decrypt-env';
+
+export default class LambdaHandler {
+  get eventType() {
+    return 'GenericEvent';
+  }
+
+  /* BEGIN LIFECYCLE METHODS */
+  validateEvent(event) {
+    return event;
+  }
+
+  async before(event, context) {
+  }
+
+  async run(event, context) {
+    throw new Error('Run method must be implemented in handler subclass');
+  }
+
+  async after(event, context) {
+  }
+  /* END LIFECYCLE METHODS */
+
+  processEvent(event) {
+    return event;
+  }
+
+  processResponse(result) {
+    return result;
+  }
+
+  processError(err) {
+    return err;
+  }
+
+  async execute(event, context) {
+    let validatedEvent = this.processEvent(event);
+    try {
+      await env.decrypt();
+
+      validatedEvent = this.validateEvent(event);
+
+      await this.before(validatedEvent, context);
+      const res = await this.run(validatedEvent, context);
+
+      return this.processResponse(res);
+    } catch (err) {
+      console.error(err.stack || err.message || err);
+      return this.processError(err);
+    } finally {
+      await this.after(validatedEvent, context);
+    }
+  }
+
+  static create() {
+    return (event, context, done) => {
+      const handler = new this();
+
+      handler.execute(event, context)
+        .then(res => done(null, res))
+        .catch((err) => {
+          setImmediate(() => {
+            throw err;
```

So this code is a bit confusing, but is only used for testing purposes.  The goal here is to actually catch errors thrown by `done`, which I assume/hope won't happen in practice, since `done` is Amazon's code.  But when we're writing tests, `done` is a function that we write that needs to be able to throw errors to fail the test properly.

So we need to throw the error, and we need to throw it outside of the context of the promise, or else the promise will swallow it and the test won't fail when it should.  Not that a test ever _should_ fail, but they need the ability to fail when appropriate to be useful.